### PR TITLE
6 сalculate claim should be based on contributor class

### PR DESCRIPTION
**Pull request by decadanse**
error when checking tokenFunder.class existing

### CODE REVIEW

File: contracts/seed/Seed.sol [hunk 1]
```diff
@@ -525,12 +526,16 @@ contract Seed {
             return 0;
         }
 
+        uint8 currentId = tokenFunder.class;
+        uint256 currentVestingDuration;
+        (,,,currentVestingDuration,) = getClass(currentId);
```
Comment (BritikovKI) [inline contracts/seed/Seed.sol:531]: You can and should get class directly, by calling array `classes[currentId]`

Comment (BritikovKI) [reply contracts/seed/Seed.sol:531]:
```suggestion
        uint256 currentVestingDuration = classes[currentId].vestingDuration; 
```

File: test/seed.js [hunk 2]
```diff
@@ -430,6 +430,7 @@ describe("Contract: Seed", async () => {
                 "0.01",
                 parseInt(fundingTokenDecimal) - parseInt(seedTokenDecimal) + 18
             ).toString();
+            // permissionedSeed = true;
```
Comment (BritikovKI) [inline test/seed.js:433]: Remove this comment pls

File: test/seed.js [hunk 3]
```diff
@@ -530,10 +540,11 @@ describe("Contract: Seed", async () => {
           await time.increase(tenDaysInSeconds);
           const claim = await setup.seed.calculateClaim(buyer1.address);
           const vestingStartTime = await setup.seed.vestingStartTime();
+          const dividor = 10000000;
           const expectedClaim = (await time.latest())
               .sub(new BN(vestingStartTime.toNumber()))
               .mul(new BN(buySeedAmount).mul(new BN(twoBN)))
-              .div(new BN(vestingDuration));
+              .div(new BN(dividor));
```
Comment (BritikovKI) [inline test/seed.js:547]: What does dividor means?

File: test/seed.js [hunk 4]
```diff
@@ -783,7 +807,10 @@ describe("Contract: Seed", async () => {
               beneficiary.address
           );
 
-          const claimTemp = new BN(buySeedAmount).mul(new BN(twoBN)).toString();
+          // amountClaimable 1020000000 --> 10200000000000000/1020000000 = 1000000000
+          const dividor = 1000000000;
```
Comment (BritikovKI) [inline test/seed.js:811]: I suppose this should be divisor?

File: test/seed.js [hunk 5]
```diff
@@ -805,23 +832,38 @@ describe("Contract: Seed", async () => {
...
+          expect(dividedFeeAmountRequired.toString()).to.equal(feeClaimed.toString());
         });
         it("funds DAO with all the fee", async () => {
           // get fundingAmount and calculate fee here
           const fee = await setup.data.seed.feeForFunder(buyer2.address);
+          const dividor = new BN(1000000000);
+          const dividedFeeAmountRequired = ethers.BigNumber.from((fee/dividor).toString());
+          const sdpb = ethers.BigNumber.from(setup.data.prevBalance);
+          const sum = ethers.BigNumber.from(dividedFeeAmountRequired.add(sdpb));
+          /*
```
Comment (BritikovKI) [inline test/seed.js:858]: Remove those comments pls

**Review by arseneeth — APPROVED** [review]

**Review by BritikovKI — APPROVED** [review]
Can you pls connect issue you're going to close before meging(take a look into my PR)
Lgtmt overall

**Pull request merged**